app/src/main/java/amazingme/app/UserContext.java

```diff
@@ -37,25 +37,21 @@ public void setParent(Parent parent) {
         this.parent = parent;
     }
 
-    public List<Child> getChildren() {
+    public Map<String, Child> getChildren() {
         return children;
     }
 
-    public List<GameResult> getGameResults() {
-        return gameResults;
-    }
+    public void setChildren(Map<String, Child> children) { this.children = children; }
 
-    public void setGameResults(List<GameResult> gameResults) {
-        this.gameResults = gameResults;
+    public void setCurrentChildUser(int currentChildUser) {
+        this.currentChildUser = currentChildUser;
     }
 
-    public void addGameResults(List<GameResult> newGameResults) { this.gameResults.addAll(newGameResults);}
+    public int getCurrentChildUser() {return this.currentChildUser; }
 
-    public int getCurrentChildUser() {
-        return currentChildUser;
+    public void addGameResultsForCurrentChildUser(List<GameResult> newGameResults) {
+        this.currentChildUser().addToGameResults(newGameResults);
     }
 
-    public void setCurrentChildUser(int currentChildUser) {
-        this.currentChildUser = currentChildUser;
-    }
+    public Child currentChildUser() { return this.children.get(childrenMapToken + String.valueOf(currentChildUser)); }
```

Comment:
this potentially return null (if there's no children) and crashes the app. we can ignore it or add a check, your call.

Reply:
the only case there should be no children is if the app doesn't load properly or if setup is incomplete. 

both of those sound like problems i'd rather fix than adding error checking... hmmmm but thinking a bit more it might not be bad... like if we've got the ability to remove a child and their data, perhaps we could have a problem...

thoughts?

Reply:
ya... i'd rather have a check elsewhere that's like "if currentChildUser == null -> do something showing the app is not in the right state" or something. especially since this is the usercontext class. i'd rather it just do it's job and let UI alert the user if there is a problem with that information. disabling pages would be a great thing to do 

so, for instance, if in the settings page we were to let them delete kids from the app. so they delete all of their kids and that's fine. don't let them see the performance page or navigate to games (or if we allow them to navigate to the game page, when we call to update results, we check and see if there are any kids to update... in this case, there wouldn't be so we do nothing)

Reply:
yeah so this is with regards to the"saveusercontext" discussion above, like if it's not saved properly (e.g. user closes app before finishing registration), then there is no child (or any other info) in it. Ideally, upon launching the app, we check for data corruption like that, and if it exists, user should be prompted with the registration pages again (PCP, child registration, etc.). For now, this is good.